critiquebrainz/frontend/external/musicbrainz_db/helpers.py

```diff
@@ -23,31 +24,31 @@ def entity_relation_helper(db, target_type, source_type, source_entity_id):
         options(joinedload("link", innerjoin=True)).\
         options(joinedload("link.link_type", innerjoin=True))
     if relation.entity0.property.mapper.class_ == relation.entity1.property.mapper.class_:
-        links = get_relationship_links(relation, query, "entity0", "entity1", target_model, source_entity_id)
-        links.extend(get_relationship_links(relation, query, "entity1", "entity0", target_model, source_entity_id))
+        get_relationship_links(relation, query, "entity0", "entity1", target_type, source_entity_ids, includes_data)
+        get_relationship_links(relation, query, "entity1", "entity0", target_type, source_entity_ids, includes_data)
     else:
         if source_model == relation.entity0.property.mapper.class_:
-            links = get_relationship_links(relation, query, "entity0", "entity1", target_model, source_entity_id)
+            get_relationship_links(relation, query, "entity0", "entity1", target_type, source_entity_ids, includes_data)
         else:
-            links = get_relationship_links(relation, query, "entity1", "entity0", target_model, source_entity_id)
-    return links
+            get_relationship_links(relation, query, "entity1", "entity0", target_type, source_entity_ids, includes_data)
 
 
-def get_relationship_links(relation, query, source_attr, target_attr, target_model, source_entity_id):
+def get_relationship_links(relation, query, source_attr, target_attr, target_type, source_entity_ids, includes_data):
     """Get relationship links between two entities.
 
     Args:
         relation (mbdata.model): Model relating the two entities.
         query (Session.query): Query object.
         source_attr (str): 'entity0' or 'entity1' based on which represents source model in relation table.
         target_attr (str): 'entity0' or 'entity1' based on which represents target model in relation table.
-        target_model (mbdata.model): Model of the target entity.
-        source_entity_id (int): ID of the source entity.
-
-    Returns:
-        List of objects of relationships between entities.
+        target_type (str): Type of the target entity.
+        source_entity_ids (list): IDs of the source entity.
+        includes_data (dict): Dictionary containing the includes data of entities.
     """
     source_id_attr = source_attr + "_id"
-    query = query.filter(getattr(relation, source_id_attr) == source_entity_id)
+    query = query.filter(getattr(relation, source_id_attr).in_(source_entity_ids))
     query = query.options(joinedload(target_attr, innerjoin=True))
-    return query.all()
+    relation_type = target_type + "-rels"
+    for link in query:
+        includes_data.setdefault(getattr(link, source_id_attr), {}).setdefault('relationship_objs', {}).\
```

Comment:
Sometimes it's useful to use `collections.defaultdict` for things like this, where you want to be able to add arbitrary nested objects. Note that this might not serialise nicely in your redis cache and so you might need to convert it before you cache it.